strawberryfields/api/connection.py

```diff
@@ -301,12 +305,43 @@ def ping(self) -> bool:
             bool: ``True`` if the connection is successful, and ``False`` otherwise
         """
         path = "/healthz"
-        response = requests.get(self._url(path), headers=self._headers)
+        response = self._request("GET", self._url(path), headers=self._headers)
         return response.status_code == 200
 
     def _url(self, path: str) -> str:
         return self._base_url + path
 
+    def _refresh_access_token(self):
+        """ Use the offline token to request a new access token """
+        self._headers.pop("Authorization", None)
+        # TODO: Make sure this is the right path
+        path = "/auth/token"
+        headers = {**self._headers}
```

Comment:
Perhaps better to do the following instead of the pop above?
```suggestion
        headers = {**self._headers}
        headers.pop("Authorization", None)
```

It feels slightly 'safer' than mutating the instance attribute above, even if it has the same effect

Reply:
This way, if the method fails midway, there are no side effects

Reply:
My thought here was that since we are in the refresh token method, we already know the existing Authorization header is invalid, so removing it from the instance headers is a housekeeping action before we replace it with a new one. Whether the method fails midway or not, any future requests are still going to fail regardless of if we've popped the header or not.